datafusion/expr/src/binary_rule.rs

```diff
@@ -153,7 +153,7 @@ fn bitwise_coercion(left_type: &DataType, right_type: &DataType) -> Option<DataT
         return None;
     }
 
-    if left_type == right_type && !is_dictionary(left_type) {
+    if left_type == right_type {
```

Comment:
this is_dictionary() call would never be true anyway due to the checks above. probably copy/paste duplication?